src/components/molecules/Visualizer/useWidgetAlignSystem.ts

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useState, useCallback } from "react";
+
+import type { WidgetLocationOptions } from "./Plugin/types";
+import type {
+  Widget,
+  WidgetAlignSystem,
+  WidgetSection,
+  WidgetZone,
+  WidgetArea,
+} from "./WidgetAlignSystem";
+
+export default ({ alignSystem }: { alignSystem: WidgetAlignSystem | undefined }) => {
+  const [overriddenAlignSystem, setOverrideAlignSystem] = useState<WidgetAlignSystem | undefined>(
+    alignSystem,
+  );
+
+  const moveWidget = useCallback((widgetId: string, options: WidgetLocationOptions) => {
+    if (
+      !widgetId ||
```

Comment:
Right now widgetId can't be undefined, so this is not necessary. If it should be string | undefined, please update it. If not, remove this check

Reply:
Yes you're right. Removed the check.